lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,483 @@
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the nowcast.lightning.NowcastLightning plugin."""
+
+
+import unittest
+
+from iris.util import squeeze
+from iris.coords import DimCoord
+from iris.cube import Cube, CubeList
+from iris.tests import IrisTest
+from iris.exceptions import CoordinateNotFoundError
+import numpy as np
+import cf_units
+
+from improver.nowcasting.lightning import NowcastLightning as Plugin
+from improver.tests.nbhood.nbhood.test_BaseNeighbourhoodProcessing import (
+    set_up_cube, set_up_cube_with_no_realizations)
+from improver.tests.ensemble_calibration.ensemble_calibration.helper_functions\
+    import add_forecast_reference_time_and_forecast_period
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        # Have to pass in a lambda to ensure two strings match the same
+        # function address.
+        set_lightning_thresholds = (lambda mins: mins, 0.)
+        result = str(Plugin(
+            lightning_thresholds=set_lightning_thresholds))
+        msg = ("""<NowcastLightning: radius={radius}, debug={debug},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+With:
+<ApplyPrecip:
+ precipitation mapping:
+   upper:  precip probability {precu} => max lightning prob {lprecu}
+   middle: precip probability {precm} => max lightning prob {lprecm}
+   lower:  precip probability {precl} => max lightning prob {lprecl}
+
+   heavy:  prob(precip>7mm/hr)  {pphvy} => min lightning prob {lprobl}
+   intense:prob(precip>35mm/hr) {ppint} => min lightning prob {lprobu}
+>
+<ApplyIce:
+ VII (ice) mapping (kg/m2):
+   upper:  VII {viiu} => max lightning prob {lviiu}
+   middle: VII {viim} => max lightning prob {lviim}
+   lower:  VII {viil} => max lightning prob {lviil}
+>
+>""".format(radius=10000., debug=False,
+            lthru=set_lightning_thresholds[0], lthrl=0.,
+            lprobu=1., lprobl=0.25,
+            precu=0.1, precm=0.05, precl=0.0,
+            lprecu=1., lprecm=0.2, lprecl=0.0067,
+            pphvy=0.4, ppint=0.2,
+            viiu=2.0, viim=1.0,
+            viil=0.5,
+            lviiu=0.9, lviim=0.5,
+            lviil=0.1)
+              )
+        self.assertEqual(result, msg)
+
+
+class Test__process_haloes(IrisTest):
+
+    """Test the _process_haloes method."""
+
+    def setUp(self):
+        """Create a cube with a single non-zero point."""
```

Comment:
Please can you document in these setUp docstrings what is in the cube that is being set up.  Ie a data array of ones with shape (...?).